### ISSUE TITLE
Evaluator should handle disconnected dispatchers

### ISSUE THREAD

**Issue by jondequinor** [equinor/ert#1606]
**Describe the bug**
We store dispatchers as `websocket` instances in a list. If, in some `recv()` call on a dispatcher, a connection-related exception is thrown, ERT will run forever, unless the dispatcher reconnects (we have no broker, so the events might be gone).

**Expected behavior**
After some time, the step for which the missing dispatcher belongs, must be set to failed with a message explaining why.

**Comment by sondreso** [comment]
After some thinking, I'm not sure we can handle this the way the expected behaviour description hints towards, where we associate a dispatcher with a certain step. Currently, we have made no guarantees that the same dispatcher will report all statuses for a step or job, meaning that an association is impossible. I think the most realistic way to solve this is through a max_runtime like behaviour, where we cancel the ensemble (with an appropriate error message) after timeout. We could have a separate timer that is lower, which indicates the time before the ensemble is canceled iff there are no connected dispatchers (ie, it's not done, but nobody seems to report statuses).

**Comment by jondequinor** [comment]
You're right. This is classical Two Generals. But I hope you mean timing out a realisation? Cancelling a whole ensemble because one dispatcher fails is excessive.

**Comment by sondreso** [comment]
Yes, I mean setting the non-finished realizations to failed after some timeout when there are no connected dispatchers. I guess the responsibility of doing this should reside in the evaluator, which means that the evaluator should be able to issue commands to the ensemble to cancel individual realizations, such that it can clean up after itself on the cluster.

**Comment by jondequinor** [comment]
Maybe this isn't a big risk, since step-level events are reported via localhost by the queue adaptor. If the job_dispatchers event reporting fails, it should not cause a deadlock (given that they report that they fail). We would still need to time out in the case of a Byzantine fault (dispatcher reports being up but is actually dead), but maybe that's not critical for this first iteration.
